Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3352,6 +3344,27 @@ extension BrowserViewController: ToolbarUrlActionsDelegate {
     }
 }
 
+extension BrowserViewController: NewTabPageDelegate {
+    func navigateToInput(_ input: String, inNewTab: Bool, switchingToPrivateMode: Bool) {
```

Comment:
I copied the base of it from FavoritesDelegate:

```
protocol FavoritesDelegate: AnyObject {
    func didSelect(input: String)
    ...
```

Not sure why it passed in String (maybe for bookmarklets?) but I just kept it that way just in case. Its called based on a fav being clicked, or context menu actioned it to open in a new tab/new private tab